### DISCUSSION TITLE
Is it possible to rewrite requests/responses?

### DISCUSSION THREAD

**Discussion by postmodern** [rubycdp/ferrum#278] [Q&A] [answered]
Is it possible to rewrite network requests / responses before they are sent out or processed by the browser?

**Comment by route** [comment, marked as answer]
I think you are looking for https://github.com/rubycdp/ferrum#interceptoptions especially `request.respond` part

**Reply by postmodern** [reply]
Will that rewrite the request that is sent to the web server or override the response that the browser receives?

**Reply by route** [reply]
That's for response

**Reply by route** [reply]
For request there's a way with standard CDP [Fetch.continueRequest ](https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#method-continueRequest)

You can override: url | method | postData | headers

Here's an example in tests https://github.com/rubycdp/ferrum/commit/5271866198dad526ca898344459f1a5afa2efafb
